src/test/java/io/debezium/connector/spanner/db/metadata/SchemaRegistryTest.java

```diff
@@ -134,4 +139,20 @@ void testUpdateSchema() throws SpannerException {
         verify(asyncResultSet, atLeast(1)).getBoolean(anyInt());
         assertTrue(schemaRegistry.getAllTables().isEmpty());
     }
+
+    @Test
+    void testUpdateSchemaFromStaleTimeStampFromCheckSchema() {
+        SchemaDao schemaDao = mock(SchemaDao.class);
+        SchemaRegistry schemaRegistry = spy(new SchemaRegistry("Stream Name", schemaDao, mock(Runnable.class)));
+        TableId tableId = TableId.getTableId("Name");
+        Timestamp timestamp = Timestamp.ofTimeMicroseconds(1L);
+        List<Column> rowType = new ArrayList<>();
+        rowType.add(Column.create("name1", "BOOL", true, 1L, false, Dialect.GOOGLE_STANDARD_SQL));
+        rowType.add(Column.create("name2", "STRING", false, 1L, false, Dialect.GOOGLE_STANDARD_SQL));
+        when(schemaDao.isPostgres()).thenReturn(false);
+        schemaRegistry.updateSchemaFromStaleTimestamp(tableId, timestamp, rowType);
+        assertEquals(1, schemaRegistry.getAllTables().size());
+        assertEquals("Name", schemaRegistry.getAllTables().iterator().next().getTableName());
+        assertEquals(2, schemaRegistry.getTable(tableId).columns().size());
```

Comment:
I think you can either use getWatchedTable, or remove this line.

Reply:
Done.